src/AsyncEnumerableExtensions/IAsyncEnumerableExtensions.cs

```diff
@@ -38,6 +42,15 @@ public static Task<string> StreamToEnd(this IAsyncEnumerable<string> stream, Act
 	public static Task<ChatDoneResponseStream?> StreamToEnd(this IAsyncEnumerable<ChatResponseStream?> stream, Action<ChatResponseStream?>? itemCallback = null)
 		=> stream.StreamToEnd(new ChatResponseStreamAppender(), itemCallback);
 
+	/// <summary>
+	/// Streams a given IAsyncEnumerable of response chunks to its end and builds one single StreamingChatCompletionUpdate out of them.
+	/// </summary>
+	/// <param name="stream">The IAsyncEnumerable to stream</param>
+	/// <param name="itemCallback">An optional callback to additionally process every single item from the IAsyncEnumerable</param>
+	/// <returns>A single StreamingChatCompletionUpdate built up from every single IAsyncEnumerable item</returns>
+	public static Task<StreamingChatCompletionUpdate?> StreamToEnd(this IAsyncEnumerable<StreamingChatCompletionUpdate?> stream, Action<StreamingChatCompletionUpdate?>? itemCallback = null)
+		=> stream.StreamToEnd(new MicrosoftAi.StreamingChatCompletionUpdateAppender(), itemCallback);
+
```

Comment:
Does this belong in `IAsyncEnumerableExtensions` or would it make more sense in its own class? This introduces `Microsoft.Extensions.AI` as a dependency and you might be able to avoid that. Does it make sense to try to keep all references to `Microsoft.Extensions.AI` under the `OllamaSharp.MicrosoftAI` namespace so it can be separately packaged?

Reply:
Nice find but the dependency is coming either way with `IChatClient`. There's no way around that dependency except making an own package for the "adapter" between `IChatClient` and `OllamaApiClient`.

In fact that's how I started, but Microsoft softly pushed my in the direction of this approach (this is the way most partners do). With this, the code actually got simpler, I think I prefer this approach in meantime.

Reply:
The dependency on `Microsoft.Extensions.AI` is alright, but limiting its leakage into other classes keeps the OllamaSharp implementation fairly loosely coupled. We don't necessarily need to build an adapter package, but keeping the adapter paradigm in mind while implementing and integrating with `Microsoft.Extensions.AI` should keep us agile enough to weather the future changes such a project is sure to have.

Reply:
That's true. I'll move this method in an own static class with the other Microsoft-related classes. Thanks for the hint.